src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +138,48 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+                .filter(accident -> isInScope(accident.getLongitude(), minLongitude, maxLongitude)
+                        && isInScope(accident.getLatitude(), minLatitude, maxLatitude))
+                .collect(Collectors.toList());
+    }
+
+    private boolean isInScope(float actualValue, float minValue, float maxValue){
+        return minValue <= actualValue && actualValue <= maxValue;
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+        Map<String, Long> groupbyWeather = roadAccidentList.stream()
+                .collect(Collectors.groupingBy(RoadAccident::getWeatherConditions, Collectors.counting()));
+        return groupbyWeather.keySet().stream()
+                .sorted((w1,w2) -> groupbyWeather.get(w2).compareTo(groupbyWeather.get(w1)))
+                .limit(3)
+                .collect(Collectors.toList());
     }
 
     /**
      * count incidents by road surface conditions
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
+        return roadAccidentList.stream()
+                .collect(Collectors.groupingBy(RoadAccident::getRoadSurfaceConditions, Collectors.counting()));
     }
 
     /**
      * To match streaming operations result, return type is a java collection instead of multimap
      * @return
      */
     public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
+        Map<String, List<String>> resultMap = new HashMap<String, List<String>>();
+        Multimap<String,String> authorityToIdsMap = ArrayListMultimap.create();
+        roadAccidentList.stream().forEach(roadAccident -> authorityToIdsMap.put(roadAccident.getDistrictAuthority(), roadAccident.getAccidentId()));
+        authorityToIdsMap.keySet().stream().forEach(authority -> resultMap.put(authority, ImmutableList.copyOf(authorityToIdsMap.get(authority))));
+        return resultMap;
```

Comment:
you can make it more pretty and easier. You have half solution in getAccidentIdsGroupedByAuthority7 now. Try to use combination collect, groupingBy and mapping. Try to use method:
 Collectors.groupingBy(Function<? super T,? extends K> classifier, Supplier<M> mapFactory, Collector<? super T,A,D> downstream)